src/pending_message.cpp

```diff
@@ -60,3 +69,73 @@ void PendingMessage::SetChoiceResetColors(bool value) {
 	choice_reset_color = value;
 }
 
+std::string PendingMessage::ApplyTextInsertingCommands(std::string input, uint32_t escape_char) {
+	if (input.empty()) {
+		return std::move(input);
+	}
+
+	std::string output;
+
+	auto iter = input.data();
+	const auto end = input.data() + input.size();
+
+	auto start_copy = iter;
+	while (iter != end) {
+		auto ret = Utils::UTF8Next(iter, end);
+		if (ret.ch != escape_char) {
+			iter = ret.iter;
+			continue;
+		}
+
+		// utf8 parsing failed
+		if (ret.ch == 0) {
+			break;
+		}
+
+		output.append(start_copy, iter);
+		start_copy = iter;
+
+		iter = ret.iter;
+		if (iter == end) {
+			break;
+		}
+
+		const auto ch = *iter;
+		++iter;
+
+		if (ch == 'N' || ch == 'n') {
+			auto parse_ret = Game_Message::ParseActor(iter, end, escape_char, true);
+			iter = parse_ret.next;
+			int value = parse_ret.value;
+
+			const auto* actor = ReaderUtil::GetElement(Data::actors, value);
+			if (!actor) {
+				Output::Error("Invalid Actor Id %d in message text", value);
```

Comment:
Why "Error"?
This has no negative side-effects.
Years ago this showed a warning but was somehow lost o.O
https://github.com/EasyRPG/Player/blob/6b12808f1a701eebde5882413932a263181c4ea6/src/window_message.cpp#L535

Reply:
RPG_RT crashes in this case. Do we prefer silent warning?

Reply:
Thought about it and changed this to warning. After thinking about it, it seems pretty disruptive to crash the game and make the player lose all their progress because of 1 malformed message.

Reply:
Funny for invalid read would be some DOS-like error handler: Retry / Abort / Continue but "warn and continue" is what I consider appropriate here.